### PR TITLE
docs(test): 3397 Add E2E test documentation

### PR DESCRIPTION
**Pull request by tihuan**
- #3397 

### Reviewers
**Functional:** 
@nayib-jose-gloria 

**Readability:** 
@ebezzi 
---

This PR adds E2E test documentation!

Please view the markdown below:
https://github.com/chanzuckerberg/single-cell-data-portal/tree/thuang-test-docs/frontend/tests

Please let me know if anything is unclear, missing, and/or can be refined 🙏 

Thanks a lot!

### CODE REVIEW

**Review by nayib-jose-gloria — COMMENTED** [review]
This is great work! There is also a top-level README in the frontend directory that discusses smoke tests--can you also check to see if there's any info there that is either outdated or should be moved/integrated into this doc here? Its probably best to consolidate info in one place as much as we can

**Comment by tihuan** [comment]
Oh gosh, totally forgot about that smoke test section 😆 It's probably at least a year outdated, if not more 🤦 I'll add a link from that README to this new one, so we have one single source of truth! Thanks for pointing that out!

**Comment by tihuan** [comment]
@nayib-jose-gloria done!

Screen Shot 2022-10-19 at 3 20 06 PM

File: frontend/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+   E.g., since tests run a lot slower in GHA vs. locally due to several reasons, such as not as powerful machine, long GHA test queue, and that GHA can't run tests in parallel in the same machine (not many CPUs to utilize), it's sometimes just a lot faster to run that one failed test locally against the same deployed environment
+
+2. A GHA E2E test against a deployed environment failed, and you want to debug the test code.
+
+   E.g., sometimes a test failure is only reproducible in a deployed environment, in such case, you have no choice but to debug the test against the deployed environment
+
+### Where
+
+All the E2E test commands can be found in `frontend/Makefile` and `frontend/package.json`. The `frontend/Makefile` commands are wrappers of `frontend/package.json` commands, so you can use either.
```
Comment (nayib-jose-gloria) [inline frontend/tests/README.md:37]: what do you think of cleaning up the makefile / package.json a bit actually? there are a few commands that seem redundant and a bit confusing to me at a glance (i.e. what is 'smoke-test-prod-build' vs e2e-prod?)

Comment (nayib-jose-gloria) [reply frontend/tests/README.md:37]: Also happy to make this a separate follow-up ticket if you agree it makes sense but prefer to tackle it separately

File: frontend/tests/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
+# Tests documentation
+
+## How to run E2E tests locally
+
+### What
+
+Running E2E tests locally means the tests are run in your local machine against the web app that's either running locally as well (localhost, docker) OR a deployed environment (dev, staging, prod)
```
Comment (metakuni) [inline frontend/tests/README.md:7]: deployed environment including rdev?

Comment (tihuan) [reply frontend/tests/README.md:7]: Ooh yes! Let me add a section for testing against an rdev 👌 Thank you!

Comment (tihuan) [reply frontend/tests/README.md]: Done! 
Screen Shot 2022-10-19 at 7 16 32 PM

**Review by metakuni — APPROVED** [review]

**Pull request merged**